website/api/helpers/compile-markdown-docs.js

```diff
@@ -0,0 +1,169 @@
+
+    // Delete existing output from previous runs, if any.
+    await sails.helpers.fs.rmrf(path.resolve(sails.config.appPath, htmlOutputPath));
+    await sails.helpers.fs.rmrf(path.resolve(sails.config.appPath, jsMenuOutputPath));
+
+    // Compile the markdown into HTML templates and menu files.
+    // (Note that "lastModified" in the resulting jsmenu files can be misleading)
+    await new Promise((resolve, reject)=>{
+      DocTemplater().build([{
+        remote: repoUrl,
+        branch: repoBranch,
+        remoteSubPath: repoPath,
+        outputExtension: 'ejs',//« the file extension for resulting HTML files
+        htmlDirPath: htmlOutputPath,
+        jsMenuPath: jsMenuOutputPath,
+        beforeConvert: (mdString, proceed)=>{// This function is applied to each template before the markdown is converted to markup
+          // Based on the github-flavored markdown's language annotation, (e.g. ```js```) add a temporary marker to code blocks that can be parsed post-md-compilation by the `afterConvert()` lifecycle hook
+          // Note: This is an HTML comment because it is easy to over-match and "accidentally" add it underneath each code block as well (being an HTML comment ensures it doesn't show up or break anything)
+          let LANG_MARKER_PREFIX = '<!-- __LANG=%';
+          let LANG_MARKER_SUFFIX = '%__ -->';
+          let modifiedMd = mdString.replace(/(```)([a-zA-Z0-9\-]*)(\s*\n)/g, '$1\n' + LANG_MARKER_PREFIX + '$2' + LANG_MARKER_SUFFIX + '$3');
+          return proceed(undefined, modifiedMd);
+        },
+        afterConvert: (html, proceed)=>{// This function is applied to each template after the markdown is converted to markup
+
+          let modifiedHtml = html;
+
+          // Replace github emoji with unicode emojis
+          // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+          // TODO: actually sub unicode, instead of the following  (there's probably an open source lib out there to do it)
+          // modifiedHtml = html.replace(/\:white_check_mark\:/g, '<i class="sails-icon icon-plus"></i>');
+          // modifiedHtml = modifiedHtml.replace(/\:white_large_square\:/g, '<i class="sails-icon icon-minus"></i>');
+          // modifiedHtml = modifiedHtml.replace(/\:heavy_multiplication_x\:/g, '<i class="sails-icon icon-times"></i>');
+          // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+          // TODO: As equivalent concepts are identified in the Fleet docs (e.g. in the API reference), maybe bring this back:
+          // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+          // // Replace ((bubble))s with HTML
+          // modifiedHtml = modifiedHtml.replace(/\(\(([^())]*)\)\)/g, '<bubble type="$1" class="colors"><span is="bubble-heart"></span></bubble>');
+          // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```

Comment:
@noahtalerman fyi


[example](https://sailsjs.com/documentation/reference/blueprint-api/find-one):
![image](https://user-images.githubusercontent.com/618009/116956330-a8905f00-ac5a-11eb-83cd-b7d6cc0af825.png)

Reply:
Just that it's there if we want to use it for our own types, so you can see how the HTML would be generated.  It's commented out right now, but if we turn this on, then any time `((string))` appears in markdown, it would show up in HTML as `<bubble type="string" class="colors"><span is="bubble-heart"></span></bubble>`  (or whatever we change the above code to when uncommenting it in the future)